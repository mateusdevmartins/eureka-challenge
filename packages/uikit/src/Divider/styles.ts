import styled from 'styled-components';

export const DividerView = styled.div`
  height: 2px;
  background-color: #d7dbe7;
  display: block;
`;
