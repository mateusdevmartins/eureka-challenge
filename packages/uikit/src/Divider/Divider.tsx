import { DividerView } from './styles';

export interface DividerProps {
  margin?: string;
  width?: string;
}

export function Divider({ margin, width }: DividerProps) {
  return (
    <DividerView
      data-testid="divider"
      style={{
        width: width ? width : '90%',
        marginTop: margin ? margin : '0px',
        marginBottom: margin ? margin : '0px',
      }}
    />
  );
}
