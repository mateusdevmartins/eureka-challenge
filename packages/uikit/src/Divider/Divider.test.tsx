import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Divider } from './Divider';

describe('Divider ', () => {
  const props = {
    margin: '24px',
    width: '100%',
  };

  it('should match snapshot', () => {
    const component = renderer.create(<Divider {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('behavior', () => {
    const components = render(<Divider {...props} />);
    const { getByTestId } = components;

    expect(getByTestId('divider').style.marginTop).toBe(props.margin);
    expect(getByTestId('divider').style.marginBottom).toBe(props.margin);
    expect(getByTestId('divider').style.width).toBe(props.width);
  });
});
