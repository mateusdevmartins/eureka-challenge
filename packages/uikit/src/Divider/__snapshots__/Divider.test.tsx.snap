// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="styles__DividerView-sc-1g8287f-0 epQbWn"
  data-testid="divider"
  style={
    Object {
      "marginBottom": "24px",
      "marginTop": "24px",
      "width": "100%",
    }
  }
/>
`;

exports[`Avatar  should match snapshot 1`] = `
<div
  className="styles__DividerView-sc-1g8287f-0 epQbWn"
  data-testid="divider"
  style={
    Object {
      "marginBottom": "24px",
      "marginTop": "24px",
      "width": "100%",
    }
  }
/>
`;

exports[`Avatar should match snapshot 1`] = `
<div
  className="styles__DividerView-sc-1g8287f-0 epQbWn"
  style={
    Object {
      "marginBottom": "24px",
      "marginTop": "24px",
      "width": "100%",
    }
  }
/>
`;
